src_c/transform.c

```diff
@@ -716,6 +1034,158 @@ surf_rotate(PyObject *self, PyObject *args, PyObject *kwargs)
+    SURF_INIT_CHECK(surf)
+    if (!dest_surface) {
+        if (adjust_size) {
+            start = MIN(MIN(x1, x2), MIN(x3, x4));
+            width = MAX(MAX(x1, x2), MAX(x3, x4)) - start + 1;
+            top = MIN(MIN(y1, y2), MIN(y3, y4));
+            height = MAX(MAX(y1, y2), MAX(y3, y4)) - top + 1;
+            newsurf = newsurf_fromsurf(surf, width, height);
+            bounds_check = 0;
+        }
+        else {
+            newsurf = newsurf_fromsurf(surf, surf->w, surf->h);
+        }
+    }
+    else {
+        newsurf = pgSurface_AsSurface(dest_surface);
+    }
+    if (!newsurf)
+        return NULL;
+
+    if (surf->w < 1 || surf->h < 1) {
+        Py_INCREF(surfobj);
+        return (PyObject *)surfobj;
+    }
+
+    if (PG_SURF_BytesPerPixel(surf) != PG_SURF_BytesPerPixel(newsurf)) {
+        return RAISE(PyExc_ValueError,
+                     "source and destination formats need to be the same");
+    }
+
+    if (PG_SURF_BytesPerPixel(surf) == 0 || PG_SURF_BytesPerPixel(surf) > 4)
+        return RAISE(PyExc_ValueError,
+                     "unsupported Surface bit depth for transform");
+
+    SDL_Point points[4] = {{x1 - start, y1 - top},
+                           {x2 - start, y2 - top},
+                           {x3 - start, y3 - top},
+                           {x4 - start, y4 - top}};
+    if (bounds_check && !(_check_inside(newsurf, points[0]) &&
+                          _check_inside(newsurf, points[1]) &&
+                          _check_inside(newsurf, points[2]) &&
+                          _check_inside(newsurf, points[3]))) {
+        return RAISE(PyExc_ValueError,
+                     "points are not within destination surface");
+    }
+
+    SDL_LockSurface(newsurf);
+    SDL_LockSurface(surf);
+    pgSurface_Lock(surfobj);
+    if (colorobj == NULL || colorobj == Py_None) {
+        /* get the background color */
+        if (SDL_GetColorKey(surf, &bgcolor) != 0) {
+            switch (PG_SURF_BytesPerPixel(surf)) {
+                case 1:
+                    bgcolor = *(Uint8 *)surf->pixels;
+                    break;
+                case 2:
+                    bgcolor = *(Uint16 *)surf->pixels;
+                    break;
+                case 4:
+                    bgcolor = *(Uint32 *)surf->pixels;
+                    break;
+                default: /*case 3:*/
+#if SDL_BYTEORDER == SDL_LIL_ENDIAN
+                    bgcolor = (((Uint8 *)surf->pixels)[0]) +
+                              (((Uint8 *)surf->pixels)[1] << 8) +
+                              (((Uint8 *)surf->pixels)[2] << 16);
+#else
+                    bgcolor = (((Uint8 *)surf->pixels)[2]) +
+                              (((Uint8 *)surf->pixels)[1] << 8) +
+                              (((Uint8 *)surf->pixels)[0] << 16);
+#endif
+            }
+            bgcolor &= ~surf->format->Amask;
+        }
+    }
+    else {
+        if (_color_from_obj(colorobj, surf->format, NULL, &bgcolor))
```

Comment:
That function does not handle string colors, making the ColorLike typehint false. To correctly handle all colors you should use this pygame api function.
```suggestion
        if (!pg_MappedColorFromObj(colorobj, surf->format, &bgcolor,
                                   PG_COLOR_HANDLE_ALL))
```